Dockerfile

```diff
@@ -9,26 +9,44 @@ RUN gem install bundler
 FROM bundler as dependencies
 
 # add nodejs, yarn, and other dependencies
-RUN curl -sL https://deb.nodesource.com/setup_9.x | bash - && \
-  curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && \
-  echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list && \
-  apt-get update && apt-get upgrade -y && \
-  apt-get install --no-install-recommends -y ca-certificates nodejs yarn \
-  build-essential libpq-dev libreoffice imagemagick graphicsmagick unzip \
-  zip ghostscript vim tesseract-ocr poppler-utils libopenjp2-tools \
-  ffmpeg qt5-default libqt5webkit5-dev xvfb xauth openjdk-11-jre \
-  --fix-missing --allow-unauthenticated
+#RUN curl -sL https://deb.nodesource.com/setup_9.x | bash - && \
+#  curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && \
+#  echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list && \
+#  apt-get update && apt-get upgrade -y && \
+#  apt-get install --no-install-recommends -y ca-certificates nodejs yarn \
+#  build-essential libpq-dev libreoffice imagemagick graphicsmagick unzip \
+#  zip ghostscript vim tesseract-ocr poppler-utils libopenjp2-tools \
+#  ffmpeg qt5-default libqt5webkit5-dev xvfb xauth openjdk-11-jre \
+#  --fix-missing --allow-unauthenticated
+
+# The alpine way
+RUN apk --no-cache update && apk --no-cache upgrade && \
+  apk add --no-cache alpine-sdk nodejs imagemagick unzip ghostscript vim yarn \
+  git sqlite sqlite-dev postgresql-dev libressl libressl-dev \
+  curl libc6-compat build-base tzdata zip autoconf automake libtool texinfo
+
+# install libffi 3.2.1
+# https://github.com/libffi/libffi/archive/refs/tags/v3.2.1.tar.gz
+# https://codeload.github.com/libffi/libffi/tar.gz/refs/tags/v3.2.1
+# apk add autoconf aclocal automake libtool
+# tar -xvzpf libffi-3.2.1.tar.gz
+# ./configure --prefix=/usr/local
+RUN mkdir -p /tmp/ffi && \
+  curl -sL https://codeload.github.com/libffi/libffi/tar.gz/refs/tags/v3.2.1 \
+  | tar -xz -C /tmp/ffi && cd /tmp/ffi/libffi-3.2.1 && ./autogen.sh &&\
+  ./configure --prefix=/usr/local && make && make install && rm -rf /tmp/ffi
 
 # install FITS for file characterization
 RUN mkdir -p /opt/fits && \
-  curl -fSL -o /opt/fits.zip https://github.com/harvard-lts/fits/releases/download/1.5.0/fits-1.5.0.zip && \
-  cd /opt && unzip fits.zip -d fits/ && chmod +X fits/fits.sh
+  curl -fSL -o /opt/fits-1.0.5.zip http://projects.iq.harvard.edu/files/fits/files/fits-1.0.5.zip && \
```

Comment:
Why the `fits` regression?

Reply:
copied from SA which still uses 1.0.5 direct from Harvard. reverted back to 1.5.0 from github